Mathlib/Data/ZMod/Basic.lean

```diff
@@ -456,6 +456,21 @@ def ringEquivCongr {m n : ℕ} (h : m = n) : ZMod m ≃+* ZMod n := by
           rw [Fin.coe_cast, Fin.val_add, Fin.val_add, Fin.coe_cast, Fin.coe_cast, ← h] }
 #align zmod.ring_equiv_congr ZMod.ringEquivCongr
 
+lemma ringEquivCongr_rfl {a : ℕ} (x : ZMod a) : ZMod.ringEquivCongr rfl x = x := by
+  cases a <;> rfl
+
+lemma ringEquivCongr_symm {a b : ℕ} (hab : a = b) (x : ZMod a) (y : ZMod b)
```

Comment:
Can you state this by saying that if `h : a = b` and `h' : b = a`, then `(ZMod.ringEquivCongr h).symm = ZMod.ringEquivCongr h'`?

I am wondering if it is worth to use `h'` also in your statement (they're of course the same, but I am not sure whether `rw` is smart enough to see through proof irrelevance, and I am afraid is not).

Same for transitivity below.

Reply:
I think your suggestion doesn't make it more general because all proofs are defeq ...

Reply:
I don't claim it is more general (as you say it surely isn't): I am just not sure `rw` or whatever will always realize it.

Reply:
I think the following shows this shouldn't be a concern; `← ringEquivCongr_symm hba.symm` produces `hba.symm.symm` on the RHS and Lean has no trouble unifying it with `hba`.
```
lemma ringEquivCongr_symm {a b : ℕ} (hab : a = b) :
    (ZMod.ringEquivCongr hab).symm = ZMod.ringEquivCongr hab.symm := by
  subst hab; cases a <;> rfl

lemma ringEquivCongr_symm' {a b : ℕ} (hab : a = b) (hba : b = a) :
    (ZMod.ringEquivCongr hab).symm = ZMod.ringEquivCongr hba := by
  rw [← ringEquivCongr_symm hba.symm] -- works
```

Reply:
I was too lazy to make this test, thanks!

Reply:
The current `ringEquivCongr_symm` can be proved using my `ringEquivCongr_symm` as follows, so I'm not sure we should have it in mathlib:
```
lemma ringEquivCongr_symm'' {a b : ℕ} (hab : a = b) (x : ZMod a) (y : ZMod b) :
    y = ZMod.ringEquivCongr hab x ↔ x = ZMod.ringEquivCongr hab.symm y := by
  rw [← ringEquivCongr_symm hab, eq_comm]; symm; apply Equiv.eq_symm_apply
```

Reply:
FWIW [here](https://github.com/leanprover-community/mathlib4/blob/d6631726d30ff5fddbfd2a01d7dac9096c18ba17/Mathlib/NumberTheory/Cyclotomic/CyclotomicCharacter.lean#L198-L203) is my application.

Reply:
I've gone with `alreadydone`'s `symm`.